gtsfm/bundle/bundle_adjustment.py

```diff
@@ -2,16 +2,21 @@
 
 Authors: Xiaolong Wu, John Lambert, Ayush Baid
 """
-from typing import NamedTuple
+import os
+from typing import Any, Dict, NamedTuple
 
 import dask
 import gtsam
+import numpy as np
 from dask.delayed import Delayed
 from gtsam import GeneralSFMFactorCal3Bundler, SfmTrack, Values, symbol_shorthand
 
+import gtsfm.utils.io as io_utils
 import gtsfm.utils.logger as logger_utils
 from gtsfm.common.gtsfm_data import GtsfmData
 
+METRICS_PATH = "result_metrics"
```

Comment:
We have this hardcoded in too many places, can we specify the output directory as a command line arg? And then the metrics subdirectory could be a config.

Reply:
Thanks for the review Akshay.

We will need a fixed place to store the outputs, since Adi's React application will look in a specific place for where to load the metrics. So I believe we will have to fix this here. Although we could save it in two places, one from a config, and one to the fixed location.

Reply:
Two points: 
- If we want a fixed directory, I think it should be a constant somewhere, maybe the config file, or in the runner. And that could be used in every module. In the worst case, all modules could accept an output_dir argument, which we hardcode in the runner. The problem with each module re-defining this constant is that someone might choose to change it at some point within a module, without knowledge of the fact that this should not be changed. 
- A user of gtsfm might want output (and metrics) to be stored somewhere else. I don't think it's good practice to restrict this to a fixed folder. We could of course have fixed paths *relative* to a variable folder that is specified by the user. Is it that way already?

Reply:
Good point. Currently we were working under the assumption that people would only run the scene optimizer at a certain level of the hierarchy, so it would be a fixed relative location if they always run `python gtsfm/runner/run_scene_optimizer.py` at that level of the hierarchy.

But you're right -- the proper way to do this is to use `Path(__file__).resolve().parent.parent.parent`, so I've updated it per that suggestion.

Good point about wanting to dump output to a specified directory -- that makes sense for experiments. I think we will have to dump in two different places, though, since Adi will need the data dumped to a specific location for React to load it.

Let me know what you all think about that @ayushbaid @akshay-krishnan .

Reply:
Looks good to me for now